compiler/renderer.nim

```diff
@@ -1680,6 +1680,10 @@ proc quoteExpr*(a: string): string {.inline.} =
   ## can be used for quoting expressions in error msgs.
   "'" & a & "'"
 
+proc doubleQuoteExpr*(a: string): string {.inline.} =
```

Comment:
why double quotes instead of single quotes?
that would mean most messages would need to be changed to a less readable syntax:
```nim
before:
expected '$1' got '$2'

after:
expected ''$1'' got ''$2''
```

If you're worried about corner cases where `a` contains a `'`, the fix is simply to make `quoteExpr` escape `'` so that `colorError` handles these correctly.

Reply:
Well `''word''` is for keeping the `'` in the error message and `'word'` is for highlighting word, to be more to the point. So I figured using quotes and double single quotes for achieving this would be best, but i suppose if the word highlighting is wanted it can be thrown away.

Reply:
I think it's simpler if error msgs stay simple (single quotes, as currently done), and the renderer can, depending on `-d:nimColorErrors`, either return input as is or transform colorize inputs in single quotes and drop the quotes in the process, which become redundant.

A special case can be made for `''` (ie when expression inside quotes is empty) to render as is, because with colors it would otherwise not show anything.

Reply:
Well it's certainly simpler, so will go back to previous state, should I also use `-d:nimNewLineOverloadError" for opt-in spacing out the overload error(Can be seen in the 2 images)

Reply:
> -d:nimNewLineOverloadError

meh on this... in any case this seems orthogonal to this PR :)
(I'm all for customizations but it should be designed well, eg via user defined templates of how error messages should look like, eg: `$file:$col:$line: $msg`, see how lldb does this to customize prompt/etc settings in ~/.lldbinit:
```
settings set frame-format frame #${frame.index}: ${frame.pc} start-addr:${line.start-addr} { ${module.file.basename}{ ${function.name-with-args}${function.pc-offset}}}{ at ${ansi.fg.yellow} ${line.file.fullpath}:${line.number}:${line.column} ${ansi.normal}}\n
```
=> more flexible. In any way, that's out of scope IMO